amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnector.scala

```diff
@@ -15,19 +15,42 @@ private[amqp] trait AmqpConnector {
     val factory = new ConnectionFactory
     settings match {
       case AmqpConnectionUri(uri) => factory.setUri(uri)
-      case AmqpConnectionDetails(host, port, maybeCredentials, maybeVirtualHost, sslProtocol) =>
-        factory.setHost(host)
-        factory.setPort(port)
+      case AmqpConnectionDetails(hostAndport, maybeCredentials, maybeVirtualHost, sslProtocol, requestedHeartbeat,
+          connectionTimeout, handshakeTimeout, shutdownTimeout, networkRecoveryInterval, automaticRecoveryEnabled,
+          topologyRecoveryEnabled, exceptionHandler) =>
+        if (hostAndport.size == 1) {
+          factory.setHost(hostAndport.head._1)
+          factory.setPort(hostAndport.head._2)
+        }
```

Comment:
Alright, now I understand the idea. However I still see some discrepancy between the way `hostAndPortList` is handled here and in `newConnection`. 

Here it checks if the size is one. And in newConnection it checks if the list is non empty. So if the hostAndPortList has only one element the host and port will be set here as well as passed to the `factory.newConnection` method. But if the hostAndPortList has more than one element then these will be passed only when calling `factory.newConnection`. 

Can we remove the `factory.setHost` and `factory.setPort` altogether?